docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/README.md

```diff
@@ -0,0 +1,272 @@
+# Messaging with Kakfa
+
+Kafka is a distributed, partitioned, replicated commit log service. It provides the functionality of a publish-subscribe messaging system with a unique design. Kafka mainly operates based on a topic model. A topic is a category or feed name to which records get published. Topics in Kafka are always multi-subscriber.
+
+This guide walks you through the process of messaging with Apache Kafka using Ballerina language.
+
+- [Messaging with Kafka](#Messaging-with-Kafka)
+  - [What you'll build](#What-youll-build)
+  - [Prerequisites](#Prerequisites)
+  - [Implementation](#Implementation)
+    - [Creating the project structure](#Creating-the-project-structure)
+    - [Developing the service](#Developing-the-service)
+  - [Deployment](#Deployment)
+    - [Deploying locally](#Deploying-locally)
+    - [Deploying on Docker](#Deploying-on-Docker)
+  - [Testing](#Testing)
+    - [Output](#Output)
+
+## What you'll build
+
+To understand how you can use Kafka for 'publish-subscribe' messaging, lets consider a real-world use case of a Product Management System. This product management system consists of a product admin portal which the product administrator can update the price for a product. Based on the content of the message sent to the product admin service, it is filtered to different partitions in the Kafka Topic.  
+
+## Prerequisites
+
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE
+
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: [VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), [IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+
+- [Apache Kafka](https://kafka.apache.org/downloads)
+  - Download the binary distribution and extract the contents
+
+### Optional Requirements
+
+- [Docker](https://docs.docker.com/install/)
+- [Kubernetes](https://kubernetes.io/docs/setup/)
+
+## Implementation
+
+### Creating the project structure
+
+Ballerina is a complete programming language that supports custom project structures. To implement the scenario in this guide, you can use the following package structure:
+
+```bash
+messaging-with-kafka
+├── Ballerina.toml
+└── src
+    └── product_management_system
+        ├── product_Admin_Service.bal
+        ├── consumer1.bal
+        ├── consumer2.bal
+        └── resources
+```
+
+### Developing the service
+
+Lets get started with the implementation of a Kafka Service called `product_admin_service` which acts as the message publisher. It contains an HTTP Service which is used by a product admin to update the price of a product.
```

Comment:
```suggestion
Lets get started with the implementation of a Kafka Service called `product_admin_service`, which acts as the message publisher. It contains an HTTP Service that is used by a product admin to update the price of a product.
```
Using "that" clarifies what you're talking about and is necessary for sentence completion. On the other hand, "which" merely adds additional information. You do not use a comma before "that" and always use a comma before "which."

Reply:
Done